base/server/src/main/java/org/dogtagpki/server/rest/JobService.java

```diff
@@ -32,6 +37,34 @@ public class JobService extends SubsystemService implements JobResource {
 
     public static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(JobService.class);
 
+    boolean isAdmin(Principal principal) {
+
+        if (principal == null) {
+            return false;
+        }
+
+        if (!(principal instanceof PKIPrincipal)) {
+            return false;
+        }
+
+        PKIPrincipal pkiPrincipal = (PKIPrincipal) principal;
+        List<String> roles = Arrays.asList(pkiPrincipal.getRoles());
```

Comment:
Can be shortened with pattern-matching:

```
        if (!(principal instanceof PKIPrincipal pkiPrincipal)) {
            return false;
        }

        List<String> roles = Arrays.asList(pkiPrincipal.getRoles());
```

Reply:
I think pkiPrincipal is out of scope in this case, you have to use it inside the if, else return null.

Reply:
Indeed. I considered that but didn't like lots of nesting for something simple. But actually it is OK, because the null check is redundant as `instanceof` returns false for a null operand, so we could do:
```
        if (principal instanceof PKIPrincipal pkiPrincipal) {
            List<String> roles = Arrays.asList(pkiPrincipal.getRoles());
            return roles.contains("Administrators");
        }
        return false;
```

Reply:
Sounds good, I'll update.